examples/run_nlp_trainer.py

```diff
@@ -0,0 +1,373 @@
+
+    print(f"\n --------\n SPAWNING TASK {task.upper()}\n DEVICE: {torch.cuda.current_device()}\n --------")
+
+    trainer = ft_hparams.initialize_object()
+    trainer.fit()
+    print(f"\nFINISHED TRAINING TASK {task.upper()}\n")
+    return trainer.state.current_metrics
+
+'''Get the run name and S3 and WandB instances by pre-constructing a dummy TrainerHparams instance. ''' 
+def get_training_info():
+    hp = TrainerHparams.create(cli_args=False, f='composer/yamls/models/bert-base.yaml')
+    run_name = hp.run_name
+    save_locally = True
+    wandb_logger = None
+    bucket = None
+    for l in hp.loggers:
+        if isinstance(l, ObjectStoreLogger):
+            save_locally = False
+            bucket = l.object_store.bucket
+        if isinstance(l, WandBLogger):
+            wandb_logger = l 
+    return (run_name, save_locally, bucket, wandb_logger)
+
+''' Validate NLP specific entrypoint arguments. ''' 
+def validate_args(args) -> None:
+    if args.training_scheme == 'finetune' and isinstance(args.load_path, dataclasses._MISSING_TYPE): 
+        raise ValueError('load_path to checkpoint folder must be specified if finetuning a model')
+
+    if not isinstance(args.load_path, dataclasses._MISSING_TYPE) and args.training_scheme != 'finetune':
+        raise ValueError('load_path cannot be specified unless finetuning a model')
+
+    if not args.file:
+        raise ValueError('must specify a top level file to store shared configs for the entrypoint. see TODO for an example.')
+
+''' Get TrainerHparams arguments from CLI and add NLP entrypoint specific arguments. ''' 
+def get_args():
+    parser = hp.get_argparse(TrainerHparams)
+    parser.add_argument('--training_scheme', help='training scheme used (one of "pretrain", "finetune", or "all")', choices=[ 'pretrain','finetune', 'all'], required=True)
+
+    args = parser.parse_args()
+    validate_args(args)
+    return args
+
+''' Logic for handling a pretraining job spawn based on storage and training settings.'''
+def run_pretrainer(args) -> None:
+    root_dir = os.path.join(os.path.dirname(composer.__file__), '..')
+    training_script = os.path.join(root_dir, 'examples/run_composer_trainer.py')
+    if args.save_locally:
+        args.save_folder = os.path.join(args.run_name, 'checkpoints')
+        if args.training_scheme == 'all':
+            new_save_folder = os.path.join(args.save_folder, 'pretrained')
+        else: # pretrain only
+            new_save_folder = args.save_folder
+        
+        subprocess_args = ['composer', training_script, '-f', f"{args.file}", '--save_folder', f"{new_save_folder}"]
+       
+    # saving on cloud
+    else:
+        if args.training_scheme == 'all':
+            args.save_folder = 'pretrained'
+            save_artifact_name = '{run_name}/pretrained/ep{epoch}-ba{batch}-rank{rank}-wct{total_wct}'
+            subprocess_args = ['composer', training_script, '-f', f"{args.file}" , '--save_folder', args.save_folder, 
```

Comment:
```suggestion
            subprocess_args = ['composer', training_script, '-f', args.file , '--save_folder', args.save_folder, 
```

Same theme of comment as below about not overriding the save arguments -- reuqire that the user specify it in the yaml. OK to default to a default if not specified.